packages/ceramic-core/src/dispatcher.ts

```diff
@@ -46,15 +58,24 @@ export default class Dispatcher extends EventEmitter {
     if (ipfsObj == null) {
       throw new Error(`Failed to find object for CID ${cid.toBaseEncodedString()} and path "${path}"`)
     }
-    // should skip cache
-    return ipfsObj.value
+    return cloneDeep(ipfsObj.value)
   }
 
   async publishHead (id: string, head: CID): Promise<void> {
+    if (!this._isRunning) {
+      console.error('Dispatcher has been closed')
+      return
+    }
+
     await this._ipfs.pubsub.publish(TOPIC, JSON.stringify({ typ: MsgType.UPDATE, id, cid: head.toString() }))
   }
 
   async handleMessage (message: any): Promise<void> {
+    if (!this._isRunning) {
+      console.error('Dispatcher has been closed')
```

Comment:
Shouldn't this throw an error instead of logging it?

Reply:
I think It's OK to just log it for now. We need to investigate why Ceramic doesn't close all the resources on `close()`. Let's do that in the separate story.